apps/language_models/scripts/sharded_vicuna_fp32.py

```diff
@@ -621,39 +620,36 @@ def get_sharded_model():
         )
         prompt = prompt_history.strip()
         input_ids = tokenizer(prompt).input_ids
+        tokens = input_ids
         prompt = print("Robot:", end=" ")
-        new_sentence = ""
-        for _ in range(200):
+        new_sentence = []
+        for iteration in range(200):
             original_input_ids = input_ids
             input_id_len = len(input_ids)
-            pad_len = SAMPLE_INPUT_LEN - input_id_len
-            attention_mask = torch.ones([1, input_id_len], dtype=torch.int64)
             input_ids = torch.tensor(input_ids)
             input_ids = input_ids.reshape([1, input_id_len])
-            attention_mask = torch.nn.functional.pad(
-                torch.tensor(attention_mask),
-                (0, pad_len),
-                mode="constant",
-                value=0,
-            )
 
-            if _ == 0:
+            if iteration == 0:
                 output = sharded_model.forward(input_ids, is_first=True)
             else:
                 output = sharded_model.forward(
                     input_ids, past_key_values=past_key_values, is_first=False
                 )
             logits = output["logits"]
             past_key_values = output["past_key_values"]
-            new_word = tokenizer.decode(torch.argmax(logits[:, -1, :], dim=1))
-            if new_word == "</s>":
+            new_token = int(torch.argmax(logits[:, -1, :], dim=1)[0])
+            if new_token == 2:
                 break
-            new_sentence += " " + new_word
-            next_token = torch.argmax(logits[:, input_id_len - 1, :], dim=1)
-            original_input_ids.append(next_token)
-            input_ids = [next_token]
-        print(new_sentence)
-        prompt_history += f"\n{new_sentence}\n"
-        if len(tokenizer(prompt_history).input_ids) > SAMPLE_INPUT_LEN:
+            new_sentence += [new_token]
+            tokens.append(new_token)
+            original_input_ids.append(new_token)
+            input_ids = [new_token]
+
+        for i in range(len(tokens)):
+            if type(tokens[i]) != int:
+                tokens[i] = int(tokens[i][0])
+        print(tokenizer.decode(new_sentence))
+        prompt_history += f"\n{tokenizer.decode(new_sentence)}\n"
+        if len(tokenizer(prompt_history).input_ids) > 5000:
```

Comment:
Is it actually 5000? If it is, why is the padding length only til 137?

Reply:
there's no padding anymore, still some legacy code though.  I tried to remove it with this patch.  I think we can just remove this limit entirely

Reply:
Wait is the vmfb dynamic or static? it certainly would have some limits no?